File: ProjectEuler/005/005.cpp
#include <iostream>
#include <algorithm>
#include <math.h>
#include <numeric>
using namespace std;
using ll = long long;

template<typename T>
inline T gcd(T a, T b){
    T c;
    while (b)
    {
        c = b;
        b = a % b;
        a = c;
    }
    return a;
}

ll lcm(ll a, ll b){
    ll g = gcd(a, b);
    return 1LL*a/g*b;
}

int main(){
    ll ans = 1;
    for(int i=2; i<=20; i++)
        ans = lcm(ans, i);
    cout << ans << '\n';
    return 0;
}

File: ProjectEuler/005/javascript/005.js
let ans = 1;

function gcd(a, b){
    let c;
    while(b){
        c = b;
        b = a%b;
        a = c;
    }
    return a;
}

function lcm(a, b){
    return a/gcd(a,b)*b;
}

for(let i=2; i<=20; i++){
    ans = lcm(ans, i);
}

console.log(ans);